### ISSUE TITLE
Enable function name override when deploying from store

### ISSUE THREAD

**Issue by rgee0** [openfaas/faas-cli#395]
## Expected Behaviour
When performing a `faas store deploy` enable a facility which enables the deployed function to take a different name to that contained within the store.json.  

## Current Behaviour
Currently when deploying from store the name of the function is taken from the `store.json` name attribute.

## Possible Solution
Use the existing `name` arg, or provide an additional flag.  If feasible the `name` arg would provide an more consistent user experience.

## Context
A use case might be during testing of store functions where `--url` has been used to point at a remote `store.json` and a function of the name to be deployed already exists.  The user may wish to deploy as `fn-test` so as to be be able to compare output with the original `fn`

**Comment by rgee0** [comment]
Derek add label: proposal

**Comment by ericstoekl** [comment]
I think the flag `--rename` should be used for this purpose, as the difference between the Store name and the deployed function name is confusing.

**Comment by alexellis** [comment]
I'm not keen on the semantics of `--rename`. In my opinion the `store` keyword is a modifier for `faas-cli deploy` - so all arguments should match for both commands.

Do we have a volunteer to work on this? It simply needs to take any override through for the name like we use for a manual deployment with the CLI.

**Comment by tranotheron** [comment]
I'm on it. :-)

**Comment by alexellis** [comment]
Where we have this override for a manual deployment:

```
faas-cli deploy --name echo --image functions/alpine:latest --fprocess cat --network func_functions
```

We should be able to override `faas-cli store deploy figlet` with a new name

For instance if we want two copies of Figlet running - one on a regular PC and one on an RPi or special node:

```
faas-cli store deploy figlet --name figlet-rpi --constraint "node==rpi"
```

Or another example:

```
faas-cli store deploy inception --name inception-gpu --constraint "gpu=nvida"
```

**Comment by alexellis** [comment]
Fixed in 0d53637 and scheduled for 0.6.9 release of the CLI.
